capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+     *                      "week": <int>
+     *                      "year": <int>
+     *                  },
+     *              "clickData":
+     *                  [
+     *                      {
+     *                          "restaurantName": <String>
+     *                          "week": <int>
+     *                          "year": <int>
+     *                          "numClicks": <int>
+     *                      },
+     *                      ...
+     *                  ]
+     *          }
+     *
+     *          "clickData" represents all the click data that we have.
+     *          "firstDate" represent the first time data was collected in our system.
+     */
+    public Map<String, Object> getAllPageViews() {
+        Query query = new Query("PageViews");
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
+
+        Map<String, Object> resultMap = new HashMap<>();
+
+        Map<String, Integer> firstPageView = new HashMap<>();
+        // Initialized with max value so any date will change the map at the beginning.
+        firstPageView.put("year", Integer.MAX_VALUE);
+        firstPageView.put("week", Integer.MAX_VALUE);
+
+        List<Map<String, Object>> clickData = populateClickData(results, firstPageView);
+
+        resultMap.put("firstDate", firstPageView);
+        resultMap.put("clickData", clickData);
+
+        return resultMap;
+    }
+
+    /**
+     * Parses the array of entities and populates a list of maps that contain the required information.
+     * @param entityList list of entities that we need to be turned into a map: List<Entity>
+     * @param firstPageView map of the earliest date in one of the entities: Map<String, Integer>
+     * @return returns a list of maps representative of the click data:
+     *              "clickData":
+     *                  [
+     *                      {
+     *                          "restaurantName": <String>
+     *                          "week": <int>
+     *                          "year": <int>
+     *                          "numClicks": <int>
+     *                      },
+     *                      ...
+     *                  ]
+     */
+    protected List<Map<String, Object>> populateClickData(List<Entity> entityList, Map<String, Integer> firstPageView) {
+        Integer entityYear;
+        Integer entityWeek;
+        Map<String, Object> tempClickData;
+        List<Map<String, Object>> clickData = new ArrayList<>();
+
+        for (Entity entity : entityList) {
+            entityYear = Integer.parseInt(entity.getProperty("year").toString());
+            entityWeek = Integer.parseInt(entity.getProperty("week").toString());
+
+            // Check for an update on the earliest date.
+            firstPageView = checkEarlierDate(firstPageView, entityWeek, entityYear);
+
+            tempClickData = new HashMap<>();
```

Comment:
nit: I wouldn't call this "tempClickData" - just clickData is fine... this is not temporary, it's exactly what gets persisted when you add it to the map.

Reply:
Method has been deleted.

Reply:
Method has been deleted.